### ISSUE TITLE
False positive IDE0079

### ISSUE THREAD

**Issue by drvic10k** [dotnet/roslyn#51884]
**Version Used**: Version 16.9.1

**Steps to Reproduce**:

1. Have a violation of some CA rule
2. Suppress in code with attribute

**Expected Behavior**: The suppress message is not marked as unnecessary

**Actual Behavior**: The suppress message is marked as unnecessary but if removed, the build shows the error that was suppressed

This behaviour started quite recently, but I cannot tell exactly which version.
Also it's not possible to add suppress message from the error window or from the code (in the code it's not even showing the error)

**Comment by sharwell** [comment]
@mavasani It appears IDE0079 is getting reported for a suppression of CA1506, which is a compilation end diagnostic. My initial guesses are:

* An old version of the analyzer may be in use, which is not providing the descriptors with `isReportedAtCompilationEnd: true`
* Or maybe IDE0079 doesn't account for compilation end analyzers due to their rare use?

**Comment by jimpaton-ufx** [comment]
This is happening for more than just `CA1506`. The following code generates `IDE0032` *Use auto property*:

``` csharp
namespace MyNamespace {
    public class MyClass {
        private readonly string value;

        public string Value => value;

        public MyClass(string value) {
            this.value = value;
        }
    }
}
```

Suppressing with attribute, generates `IDE0079`:

``` csharp
namespace MyNamespace {
    public class MyClass {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0032:Use auto property", Justification = "<Pending>")]
        private readonly string value;

        public string Value => value;

        public MyClass(string value) {
            this.value = value;
        }
    }
}
```

Suppressing with `#pragma` does not generate `IDE0079`:

``` csharp
namespace MyNamespace {
    public class MyClass {
#pragma warning disable IDE0032 // Use auto property
        private readonly string value;
#pragma warning restore IDE0032 // Use auto property

        public string Value => value;

        public MyClass(string value) {
            this.value = value;
        }
    }
}
```

**Comment by drvic10k** [comment]
> @mavasani It appears IDE0079 is getting reported for a suppression of CA1506, which is a compilation end diagnostic. My initial guesses are:
> 
> * An old version of the analyzer may be in use, which is not providing the descriptors with `isReportedAtCompilationEnd: true`
> * Or maybe IDE0079 doesn't account for compilation end analyzers due to their rare use?

sorry for the late response

I updated the analyzer NuGet packages and now it seems to work correctly, that means the suppress messages are not marked as unnecessary

the problem where it's not possible to add suppress messages from the error window stays, but it's possible to add from code, so that is acceptable

**Comment by mavasani** [comment]
> I updated the analyzer NuGet packages and now it seems to work correctly, that means the suppress messages are not marked as unnecessary

Thanks for confirming @drvic10k! I'll close this issue.

 > the problem where it's not possible to add suppress messages from the error window stays, but it's possible to add from code, so that is acceptable

Please feel free to file a separate issue for error list command menu for suppression not working, thanks!
